File: 351004/Nepesov/Lab1/untitled/src/main/java/org/example/config/JacksonConfig.java
package org.example.config;

import org.springframework.boot.autoconfigure.jackson.Jackson2ObjectMapperBuilderCustomizer;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class JacksonConfig {
    @Bean
    public Jackson2ObjectMapperBuilderCustomizer jsonCustomizer() {
        return builder -> {
            builder.featuresToDisable(com.fasterxml.jackson.databind.SerializationFeature.WRAP_ROOT_VALUE);
            builder.featuresToDisable(com.fasterxml.jackson.databind.DeserializationFeature.UNWRAP_ROOT_VALUE);
        };
    }
}

File: 351002/Abrashin/src/main/kotlin/com/example/distcomp/config/JacksonConfig.kt
package com.example.distcomp.config

import com.fasterxml.jackson.databind.SerializationFeature
import org.springframework.context.annotation.Configuration
import org.springframework.http.converter.HttpMessageConverter
import org.springframework.http.converter.json.MappingJackson2HttpMessageConverter
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer

@Configuration
class JacksonConfig : WebMvcConfigurer {
    override fun extendMessageConverters(converters: MutableList<HttpMessageConverter<*>>) {
        converters.filterIsInstance<MappingJackson2HttpMessageConverter>().forEach {
            it.objectMapper.disable(SerializationFeature.WRAP_ROOT_VALUE)
        }
    }
}
